builtins/davix/CMakeLists.txt

```diff
@@ -48,6 +48,8 @@ ExternalProject_Add(DAVIX
     -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT}
     -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
     -DLIB_SUFFIX=
+  BUILD_COMMAND gmake
```

Comment:
This shouldn't be set. We are using CMake for the configuration/build. The build tool is set by cmake, and the build command should be the default one, i.e. `cmake --build <BUILD_DIR>` or such, so it calls the right tool itself. What you may need instead is to add `-G "${CMAKE_GENERATOR}"` or similar to propagate what the actual generator is (could be ninja instead of make, for example).

Reply:
No, what I meant is that we shouldn't touch `BUILD_COMMAND`, but just relay down the build tool configured by ROOT itself with `-G "${CMAKE_GENERATOR}"` in the `CMAKE_ARGS` part above.

Reply:
nope, setting -j1 also causes trouble.

Reply:
@amadio adding -G ${CMAKE_GENERATOR} in the CMAKE_ARGS (instead of BUILD_COMMAND) produces the same result, probably the same piece of lgic is generating the build command anyway. Funny thing is one error comes from make, the other outputs in the error log seem to come from gmake:
```
make[6]: option requires an argument -- j
gmake[5]: *** [CMakeFiles/BuildCurlBundled.dir/build.make:86: deps/curl/src/BuildCurlBundled-stamp/BuildCurlBundled-build] Error 2
gmake[4]: *** [CMakeFiles/Makefile2:214: CMakeFiles/BuildCurlBundled.dir/all] Error 2
gmake[4]: *** Waiting for unfinished jobs....
```

Reply:
> nope, setting -j1 also causes trouble.

What do you mean by this? How are you trying to build?

Could you please update the description with the CMake configuration command you are trying to use, and attach the configuration log and `CMakeCache.txt`? Thank you!

Reply:
From what I see above, Davis is using make, but the curl being built by it internally is using gmake.

Reply:
I'm building (as described in the bug report) with cmake --build . -j4. I'll attach the files

Reply:
Actually, after starting from scratch adding BUILD_COMMAND gmake does not help, still craps out. I think it went through in my tests because I did a `make` in the build directory manually (which built davix without problems) after which the rest of root goes through after restarting with cmake --build . (until it hits #14492). Marked the PR [WIP].
So - davix seems to build fine with a manual issue of `make` (or `gmake`), it's just that cmake is trying really hard to issue the wrong build command.

Reply:
[CMakeCache.txt](https://github.com/root-project/root/files/14302959/CMakeCache.txt)
[CMakeConfigureLog.yaml.txt](https://github.com/root-project/root/files/14302964/CMakeConfigureLog.yaml.txt)
adding files

Reply:
> I'm building (as described in the bug report) with cmake --build . -j4. I'll attach the files

Sorry, I missed that this was in the description. This might be your problem. Try either `cmake --build . --parallel 4` or `cmake --build . -- -j4`. You need the `--` there to tell CMake which options to forward to the build tool.

Reply:
@amadio - indeed, that worked, thanks! Funny though it only failed there, other builtins seemed to always compile without problems. Anyway, updated the PR.

Reply:
@amadio sorry, it worked only without the parallel option (cmake --build .) with your suggestion, same error.

Reply:
What version of CMake are you using? My guess is this is likely a CMake bug. Try to configure with `-G Ninja`, maybe that will work for you.

Reply:
I was using cmake 3.28.1, upgraded to 3.28.3, same problem.